docs/development/testguide.rst

```diff
@@ -138,47 +105,29 @@ pytest
 ======
 
 The test suite can also be run directly from the native ``pytest`` command,
-which is generally faster than using tox for iterative development. In
-this case, it is important for developers to be aware that they must manually
-rebuild any extensions by running::
-
-    python -m pip install --editable ".[test]"
+which is much faster than using ``tox`` for iterative development. In
+this case, it is important to rebuild any extensions that have changed by running
+the following. This assumes you are working in an :ref:`isolated development environment<create-isolated-env>`.
 
-before running the test with pytest with::
-
-    pytest
-
-Instead of calling ``python -m pip install --editable ".[test]"``, you can also build the
-extensions with::
-
-    python setup.py build_ext --inplace
-
-which avoids also installing the developer version of astropy into your current
-environment - however note that the ``pip`` command is required if you need to
-test parts of the package that rely on certain `entry points
-<https://setuptools.readthedocs.io/en/latest/pkg_resources.html#entry-points>`_
-being installed.
+    python -m pip install --editable ".[dev_all]"
 
 It is possible to run only the tests for a particular subpackage or set of
-subpackages.  For example, to run only the ``wcs`` tests from the
+subpackages.  For example, to run only the ``wcs`` and ``utils`` tests from the
 commandline::
 
-    pytest -P wcs
-
-Or, to run only the ``wcs`` and ``utils`` tests::
-
     pytest -P wcs,utils
```

Comment:
In my own use I agree, but this ignores Windows developers.

Reply:
@pllim, is it true you can't do brace expansion on Windows?

Reply:
I never even knew about brace expansion. I will have to check when I get a chance. 😅

Reply:
Re: braces on Windows -- Works on Windows 11 but I use Git Bash within VS Code, not Powershell. @duytnguyendtn , do you still use Powershell? Would you mind running this using code checkout and installed from `main` (does not have to be this PR branch):

```
pytest {astropy,docs}/{wcs,utils} -p no:cacheprovider
```

I had to give `-p no:cacheprovider` because of Norton but if you don't have the cache dir creation problem, you don't need this option.

Thanks!

Reply:
Indeed looks like brace expansion isn't supported in Powershell, or at least there's something syntactically off; it didn't even execute:

```
(envastropypr16561) PS C:\Users\dtnguy20\Documents\duydir\gitrepos> pytest {astropy,docs}/{wcs,utils}
At line:1 char:16
+ pytest {astropy,docs}/{wcs,utils}
+                ~
Missing argument in parameter list.
At line:1 char:27
+ pytest {astropy,docs}/{wcs,utils}
+                           ~
Missing argument in parameter list.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingArgument
```

Reply:
There's a whole thread on StackOverflow on people trying to find hacky workarounds: https://stackoverflow.com/questions/16222477/powershell-equivalent-of-bash-brace-expansion-for-generating-lists-arrays

Reply:
Wow, glad I asked. Thanks for checking and the extra info!

@eerovaher , I guess we cannot accept your suggestion in using brace expansion then, at least for now.

Reply:
I am still convinced that my suggestion was sensible, but I can accept that Powershell is not.

Reply:
I do not disagree with you when it comes to Powershell and sensibility. But we're stuck in the universe where it exists...

Reply:
I can only apologize for the behavior of the windows overlords... Thank you for not forgetting about us Windows devs!

On the other hand, Powershell *is* open-source 👀: https://github.com/PowerShell/PowerShell